packages/common/src/i18n/format_date.ts

```diff
@@ -399,10 +399,18 @@ function getFirstThursdayOfYear(year: number) {
       year, 0, 1 + ((firstDayOfYear <= THURSDAY) ? THURSDAY : THURSDAY + 7) - firstDayOfYear);
 }
 
-function getThursdayThisWeek(datetime: Date) {
+/**
+ *  ISO Week starts on day 1 (Monday) and ends with day 0 (Sunday)
+ */
+export function getThursdayThisIsoWeek(datetime: Date) {
+  // getDay returns 0-6 range with sunday as 0.
+  const currentDay = datetime.getDay();
+
+  // On a sunday, we read the previous thursday
```

Comment:
I think this is missing the _why_ (or at least it's located quite far in the function's docblock):

```suggestion
  // On a Sunday, read the previous Thursday since ISO weeks start on Monday.
```